cranelift/codegen/src/isa/aarch64/abi.rs

```diff
@@ -808,8 +816,21 @@ fn is_caller_save_reg(call_conv: isa::CallConv, r: RealReg) -> bool {
             r.get_hw_encoding() <= 17
         }
         RegClass::V128 => {
-            // v0 - v7 inclusive and v16 - v31 inclusive are caller-saves.
-            r.get_hw_encoding() <= 7 || (r.get_hw_encoding() >= 16 && r.get_hw_encoding() <= 31)
+            // v0 - v7 inclusive and v16 - v31 inclusive are caller-saves. The
+            // upper 64 bits of v8 - v15 inclusive are also caller-saves.
+            // However, because we cannot currently represent partial registers
+            // to regalloc.rs, we indicate here that every vector register is
+            // caller-save. Because this function is used at *callsites*,
+            // approximating in this direction (save more than necessary) is
+            // conservative and thus safe.
+            //
```

Comment:
This all is reasonable (and as we discussed and agreed), but it has the side effect that I am not entirely happy with the fact that the name of this function doesn't make clear that it now only pertains to call sites.  Would you be prepared to consider the following changes (or something modulo them, so as to increase clarity for the reader) ?

* change name of function to `is_reg_clobbered_by_a_call_instruction`

* change name of formal param from `call_conv` to `call_conv_of_callee`  (since it's kinda ambiguous at least from the name)

Reply:
Ah, I like this much better! The `ABIMachineSpec` now has `get_regs_clobbered_by_call()`, and aarch64 has `is_reg_saved_in_prologue()`, `get_regs_saved_in_prologue()`, and `is_reg_clobbered_by_call()`.